assignment/assignment.hs

```diff
@@ -26,25 +31,39 @@ import Data.Char(intToDigit)
 -- predicate returns `True`. Implement `myfilter` using `foldl`. (Hint: you 
 -- might need to use `++` rather than `:` to build the result)
 
--- myfilter :: (a -> Bool) -> [a] -> [a]
+--myfilter :: (a -> Bool) -> [a] -> [a]
+
+addtolist fn list element
+  | (fn element == True)  = list : element
+  | otherwise             = list
 
 
+myfilter fn = foldl (addtolist fn) []
+
 -------
 -- 3 --
 -------
 -- The `and` and `or` functions take a list of booleans.
--- `and` returns true only iff all the values are true.
+-- `and` returns true only if all the values are true.
 -- `or` returns true if any are true.
 --
 -- (Side note: when implementing your functions, did you get a wavy
 -- line suggestion say that you could replace your code with the
 -- library function. Stop for a minute and see if you can work
 -- out how Haskell worked this out.)
   
--- myand:: [ Bool ] -> Bool 
+--myand:: [ Bool ] -> Bool 
 
 -- myor:: [ Bool ] -> Bool 
 
+pairwiseAnd boolVal [] = True
+pairwiseAnd boolVal (x:xs) 
+  | boolVal == False = False
+  | x == False = False
+  | otherwise = True
+
+myand = foldl (pairwiseAnd) True
+
```

Comment:
So here's a good example of my thinking of things as True or False can get in the way.

You could simplify your pairWiseAnd down by removing the equality tests, and doing a little refactoring, and you'd discover it was just `&&`, so

~~~ hs
myand:: [ Bool ] -> Bool 
myand = foldl (&&) True

myor:: [ Bool ] -> Bool 
myor = foldl (||) False
~~~